source/Halibut/Transport/SecureListener.cs

```diff
@@ -209,7 +214,7 @@ async Task ExecuteRequest(TcpClient client)
                     if (req.Substring(0, 2) != "MX")
                     {
                         log.Write(EventType.Diagnostic, "Appears to be a web browser, sending friendly HTML response");
-                        SendFriendlyHtmlPage(ssl);
+                        await SendFriendlyHtmlPage(ssl);
```

Comment:
nit/alt: you could have also approached this by leaving the existing methods alone and adding a new version e.g. `SendFriendlyHtmlPageAsync` and do this if..else.. here

That way you would not have to convert the existing method to async

Reply:
Yeah, it was a bit 6 of one, half dozen of the other tbh.

I figured if I change the existing method now, then we don't have to change it when we remove the sync code paths 🤷🏻‍♂️

Reply:
We wouldn't have to change the existing sync method when we remove sync, we would just remove it 😉